### PR TITLE
NTVEPLUGIN-405: fixed the following procedures for new Jenkins tabular to div changes: Run Procedure, Run Pipeline, Deploy Application, Trigger Release

### PR DESCRIPTION
**Pull request by ovasilko**

### CODE REVIEW

**Review by justnoxx — APPROVED** [review]

**Comment by justnoxx** [comment]
@ovasilko please let me know we can merge it

File: src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowDeployApplication/config.jelly [hunk 1]
```diff
@@ -117,20 +117,40 @@ All rights reserved.
...
-                    var row1 = table.insertRow(table.rows.length - 2);
-                    row1.className = "_ef_da_row";
-                    if (parameters.length) {
-                        row1.appendChild(createTitleForDeployApplication("Deploy Parameters"));
-                        parameters.forEach(function (elem) {
-                            var row = table.insertRow(table.rows.length - 2);
-                            row.className = "_ef_da_parameters _ef_da_row";
-                            addParameterRowForDeployApplication(row, elem.actualParameterName, elem.value, false);
-                        });
+                    if (table == null) {
```
Comment (fqueiruga) [inline src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowDeployApplication/config.jelly:120]: Where does this variable come from?

File: src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowDeployApplication/config.jelly [hunk 2]
```diff
@@ -117,20 +117,40 @@ All rights reserved.
...
+                        if (parameters.length) {
+                            var el = document.createElement("div");
+                            el.innerHTML = "&lt;br&gt;&lt;b&gt;Deploy Parameters&lt;/b&gt;";
+                            el.className = "_ef_da_row";
+                            var div = getElementByXpath("//*[@id=\"ef_da_environmentName\"]");
```
Comment (fqueiruga) [inline src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowDeployApplication/config.jelly:125]: I'd suggest replacing this with `document.getElementById="ef_da_environmentName"`. `getElementByXPath` is not native JavaScript.

File: src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowPipelinePublisher/config.jelly [hunk 3]
```diff
@@ -109,20 +109,40 @@ All rights reserved.
...
-                            var row = table.insertRow(table.rows.length - 2);
-                            row.className = "_ef_parameters _ef_row";
-                            addParameterRowForRunPipeline(row, elem.parameterName, elem.parameterValue);
-                        });
+                    if (table == null) {
+                        if (json.length) {
+                            var el = document.createElement("div");
+                            el.innerHTML = "&lt;br&gt;&lt;b&gt;Pipeline Parameters&lt;/b&gt;";
+                            el.className = "_ef_row";
+                            var div = getElementByXpath("//*[@id=\"pipelineN\"]");
```
Comment (fqueiruga) [inline src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowPipelinePublisher/config.jelly:117]: Ditto

File: src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowRunProcedure/config.jelly [hunk 4]
```diff
@@ -104,21 +104,40 @@ All rights reserved.
...
-                            var row = table.insertRow(table.rows.length - 2);
-                            row.className = "_ef_rp_parameters _ef_rp_row";
-                            addParameterRowForRunProcedure(row, elem.actualParameterName, elem.value, false);
-                        });
+                    if (table == null) {
+                        if (parameters.length) {
+                            var el = document.createElement("div");
+                            el.innerHTML = "&lt;br&gt;&lt;b&gt;Procedure Parameters&lt;/b&gt;";
+                            el.className = "_ef_rp_row";
+                            var div = getElementByXpath("//*[@id=\"ef_rp_procedureName\"]");
```
Comment (fqueiruga) [inline src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowRunProcedure/config.jelly:112]: Ditto

File: src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowTriggerRelease/config.jelly [hunk 5]
```diff
@@ -102,50 +102,83 @@ All rights reserved.
...
-                            var row = table.insertRow(table.rows.length - 2);
-                            row.className = "_ef_sr_stages _ef_st_row";
-                            addParameterRowForTriggerRelease(row, elem.stageName, elem.stageValue, true);
-                        });
+                    if (table == null) {
+                        if (parameters.length) {
+                            var el = document.createElement("div");
+                            el.innerHTML = "&lt;br&gt;&lt;b&gt;Release pipeline parameters&lt;/b&gt;";
+                            el.className = "_ef_st_row";
+                            var div = getElementByXpath("//*[@id=\"ef_sr_startingStage\"]");
```
Comment (fqueiruga) [inline src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowTriggerRelease/config.jelly:118]: As above

File: src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowTriggerRelease/config.jelly [hunk 6]
```diff
@@ -102,50 +102,83 @@ All rights reserved.
...
-                            var row = table.insertRow(table.rows.length - 2);
-                            row.className = "_ef_sr_parameters _ef_st_row";
-                            addParameterRowForTriggerRelease(row, elem.parameterName, elem.parameterValue, false);
-                        });
+                    if (table == null) {
+                        if (stages.length) {
+                            var el = document.createElement("div");
+                            el.innerHTML = "&lt;br&gt;&lt;b&gt;Stages to run&lt;/b&gt;";
+                            el.className = "_ef_st_row";
+                            var div = getElementByXpath("//*[@id=\"ef_sr_startingStage\"]");
```
Comment (fqueiruga) [inline src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowTriggerRelease/config.jelly:153]: Ditto

File: src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowDeployApplication/config.jelly [hunk 7]
```diff
@@ -117,20 +117,40 @@ All rights reserved.
...
+                                row.className = "_ef_da_parameters _ef_da_row";
+                                addParameterRowForDeployApplication(row, elem.actualParameterName, elem.value, false, true);
+                            });
+                        }
                     }
                 }
             }
         }
 
+        function insertAfter(referenceNode, newNode) {
```
Comment (fqueiruga) [inline src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowDeployApplication/config.jelly:150]: Are this functions defined on the global `window` namespace? If so, I'd suggest prefixing a name to namespace them. Maybe something like `cbcd_insertAfter` or `cbcd_ addParameterRowForDeployApplication`. Just a thought.

File: src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowDeployApplication/config.jelly (hunk 1 above)
Comment (oleksiilv) [reply src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowDeployApplication/config.jelly:120]: this was used in older version of the code (and still to be used for older versions of Jenkins). See the code above:
```
            var table = document.querySelector("[descriptorid='electricFlowDeployApplication'] table");
            if (table == null) {
                table = getElementByXpath("//select[@checkdependson='validationTrigger']/ancestor::table[@name]");
            }
```

File: src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowDeployApplication/config.jelly (hunk 2 above)
Comment (oleksiilv) [reply src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowDeployApplication/config.jelly]: thanks, updated accordingly. PS. we still use getElementByXpath in some places

File: src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowPipelinePublisher/config.jelly (hunk 3 above)
Comment (oleksiilv) [reply src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowPipelinePublisher/config.jelly]: updated

File: src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowRunProcedure/config.jelly (hunk 4 above)
Comment (oleksiilv) [reply src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowRunProcedure/config.jelly]: updated

File: src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowTriggerRelease/config.jelly (hunk 5 above)
Comment (oleksiilv) [reply src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowTriggerRelease/config.jelly]: updated

File: src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowTriggerRelease/config.jelly (hunk 6 above)
Comment (oleksiilv) [reply src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowTriggerRelease/config.jelly]: updated

File: src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowDeployApplication/config.jelly (hunk 7 above)
Comment (oleksiilv) [reply src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowDeployApplication/config.jelly]: thanks, updated name of insertAfter, for other existing functions we will do renaming separately.

File: src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowDeployApplication/config.jelly (hunk 1 above)
Comment (ovasilko) [reply src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowDeployApplication/config.jelly:120]: this was used in older version of the code (and still to be used for older versions of Jenkins). See the code above:
```
            var table = document.querySelector("[descriptorid='electricFlowDeployApplication'] table");
            if (table == null) {
                table = getElementByXpath("//select[@checkdependson='validationTrigger']/ancestor::table[@name]");
            }
```

File: src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowDeployApplication/config.jelly (hunk 2 above)
Comment (ovasilko) [reply src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowDeployApplication/config.jelly]: thanks, updated accordingly. PS. we still use getElementByXpath in some places

File: src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowPipelinePublisher/config.jelly (hunk 3 above)
Comment (ovasilko) [reply src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowPipelinePublisher/config.jelly]: updated

File: src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowRunProcedure/config.jelly (hunk 4 above)
Comment (ovasilko) [reply src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowRunProcedure/config.jelly]: updated

File: src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowTriggerRelease/config.jelly (hunk 5 above)
Comment (ovasilko) [reply src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowTriggerRelease/config.jelly]: updated

File: src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowTriggerRelease/config.jelly (hunk 6 above)
Comment (ovasilko) [reply src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowTriggerRelease/config.jelly]: updated

File: src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowDeployApplication/config.jelly (hunk 7 above)
Comment (ovasilko) [reply src/main/resources/org/jenkinsci/plugins/electricflow/ElectricFlowDeployApplication/config.jelly]: thanks, updated name of insertAfter, for other existing functions we will do renaming separately.

**Review by fqueiruga — APPROVED** [review]

**Comment by fqueiruga** [comment]
Approving based on how the code looks. I have not performed any sort of testing on it.

**Comment by justnoxx** [comment]
Merging it because @fqueiruga has approved it.

**Pull request merged**